src/main/java/com/arpnetworking/clusteraggregator/GuiceModule.java

```diff
@@ -314,6 +323,24 @@ private ActorRef provideJvmMetricsCollector(final ActorSystem system, final Metr
         return system.actorOf(JvmMetricsCollector.props(_configuration.getJvmMetricsCollectionInterval(), metricsFactory));
     }
 
+    @Provides
+    @Singleton
+    @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD") // Invoked reflectively by Guice
+    private PeriodicMetrics providePeriodicMetrics(final MetricsFactory metricsFactory, final LifecycleRegistration lifecycle) {
+        final TsdPeriodicMetrics periodicMetrics = new TsdPeriodicMetrics.Builder()
+                .setMetricsFactory(metricsFactory)
+                .build();
+        final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor(
+                r -> new Thread(r, "PeriodicMetricsCloser"));
+        final long offsetMillis = 1250 - (System.currentTimeMillis() % 1000);;
```

Comment:
nit. extra semicolon.

Where does the 1250 come from? If the goal is to align on the second shouldn't it be 1000?

Reply:
The goal is to align on 250ms past the second. Ending the collection too close to the second boundary can result in samples flipping between time buckets. There are more comprehensive ways to do this. However, this is a reasonable incremental improvement in my opinion.

Reply:
That makes sense. A comment to that effect would be appreciated!